src/c++/perf_analyzer/metrics_manager.cc

```diff
@@ -62,11 +62,7 @@ MetricsManager::QueryMetricsEveryNMilliseconds()
     Metrics metrics{};
     clientbackend::Error err{client_backend_->Metrics(metrics)};
     if (err.IsOk() == false) {
-      // FIXME: this custom logic enables PA in C API mode, remove in TMA-775
-      if (err.Message() !=
-          "client backend of kind TRITON_C_API does not support Metrics API") {
-        throw std::runtime_error(err.Message());
-      }
+      throw std::runtime_error(err.Message());
     }
 
     CheckForMissingMetrics(metrics);
```

Comment:
Won't this function only ever go through once? The `has_given_missing_metrics_warning_` looks to always be set after the first call, whether any warnings occurred or not.

Reply:
Nice catch--fixed that logic issue 👍